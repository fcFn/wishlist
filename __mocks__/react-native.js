const RN = require('react-native');

RN.AppState.removeEventListener = function() {};
RN.BackAndroid.removeEventListener = jest.fn();

module.exports = RN;
