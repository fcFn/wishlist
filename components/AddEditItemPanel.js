// This component is used for adding and editing items

import React, {Component} from 'react';
import {View, TextInput} from 'react-native';
import makeTestIDs from '../helpers/misc';

export default class AddEditItemPanel extends Component {
  componentWillUnmount() {
    this.textInput.blur();
    this.textInput.clear();
  }

  render() {
    return <View>
      <TextInput
      ref={(input) => this.textInput = input}
      defaultValue={this.props.placeholder}
      autoFocus={true}
      autoCapitalize={'sentences'}
      selectTextOnFocus={true}
      style={this.props.textInputStyle}
      onSubmitEditing={this.props.onSubmitEditing}
      {...makeTestIDs('AddEditItemTextInput')}
    />
    </View>;
  }
}
