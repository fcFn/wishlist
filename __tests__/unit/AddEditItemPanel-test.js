import React from 'react';
import 'react-native';
import AddEditItemPanel from '../../components/AddEditItemPanel';
import renderer from 'react-test-renderer';

describe('Item adding and editing input component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<AddEditItemPanel />);
    expect(tree).toMatchSnapshot();
  });
});
