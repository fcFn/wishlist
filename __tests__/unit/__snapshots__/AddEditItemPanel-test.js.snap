// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item adding and editing input component renders correctly 1`] = `
<View>
  <TextInput
    accessibilityLabel="AddEditItemTextInput"
    autoCapitalize="sentences"
    autoFocus={true}
    defaultValue={undefined}
    onSubmitEditing={undefined}
    selectTextOnFocus={true}
    style={undefined}
    testID="AddEditItemTextInput"
  />
</View>
`;
